packages/taquito-core/src/errors.ts

```diff
@@ -0,0 +1,254 @@
+// ==========================================================================================
+// parent error classes for Taquito
+// ==========================================================================================
+/**
+ *  @category Error
+ *  @description Parent error class all taquito errors to extend from
+ */
+export class TaquitoError extends Error {}
+
+/**
+ *  @category Error
+ *  @description Error indicates invalid user inputs
+ */
+export class ParameterValidationError extends TaquitoError {}
+
+/**
+ *  @category Error
+ *  @description Error returned by RPC node
+ */
+export class RpcError extends TaquitoError {}
+
+/**
+ *  @category Error
+ *  @description Error indicates TezosToolKit has not been configured appropriately
+ */
+export class TezosToolkitConfigError extends TaquitoError {}
+
+/**
+ *  @category Error
+ *  @description Error indicates a requested action is not supported by Taquito
+ */
+export class UnsupportedAction extends TaquitoError {}
```

Comment:
Other error classes end with `Error`. It might be a good practice to keep this as a convention and apply it to this one as well.